Tests/OpenCombineTests/Helpers/ExecuteOnBackgroundThread.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  ExecuteOnBackgroundThread.swift
+//  
+//
+//  Created by Sergej Jaskiewicz on 04.02.2020.
+//
+
+#if canImport(Darwin)
+import Darwin
+#elseif canImport(Glibc)
+import Glibc
+#else
+#error("How to do threads on this platform?")
+#endif
+
+#if canImport(Darwin)
+private typealias ThreadPtr = UnsafeMutablePointer<pthread_t?>
+#else
+private typealias ThreadPtr = UnsafeMutablePointer<pthread_t>
+#endif
+
+func executeOnBackgroundThread<ResultType>(
+    _ body: () -> ResultType
+) -> ResultType {
+    return withoutActuallyEscaping(body) { body in
+
+        // We need this because @convention(c) closures can't capture generic params.
+        var typeErasedBody: () -> UnsafeMutableRawPointer = {
+            let resultPtr = UnsafeMutablePointer<ResultType>.allocate(capacity: 1)
+            resultPtr.initialize(to: body())
+            return UnsafeMutableRawPointer(resultPtr)
+        }
+
+        return withUnsafeMutablePointer(to: &typeErasedBody) { typeErasedBody in
+            let _backgroundThread = ThreadPtr.allocate(capacity: 1)
+
+            defer { _backgroundThread.deallocate() }
+
+            var status: Int32 = 0
+
+            // We could use Foundation's Thread, but it doesn't work on Linux for some
+            // reason.
+            status = pthread_create(
```

Comment:
@broadwaylamb small question: why you didn't set BG QoS for the thread:

```swift
 var qosAttribute = pthread_attr_t()
      pthread_attr_init(&qosAttribute)
      pthread_attr_set_qos_class_np(&qosAttribute, QOS_CLASS_BACKGROUND, 0)
      
      status = pthread_create(
        _backgroundThread,
        &qosAttribute,
        { context in
          
          return context
            .assumingMemoryBound(to: (() -> UnsafeMutableRawPointer).self)
            .pointee()
      },
        typeErasedBody
      )
....
```

Reply:
@ibrahimkteish why?

Reply:
Fair enough 👍, when do you see that needed ?